src/xmipp/libraries/reconstruction/volume_subtraction.cpp

```diff
@@ -170,14 +170,17 @@ void POCSFourierAmplitudeRadAvg(MultidimArray<std::complex<double>> &V,
 			{
 				FFT_IDX2DIGFREQ_FAST(j,V1size_x,V1size2_x,V1sizei_x,wx)
 				double w = sqrt(wx*wx + wy2_wz2);
-				auto iw = (int)round(w*V1size_x);
+				auto iw = (int)floor(w*V1size_x);
```

Comment:
it would be safer to do
`auto iw = std::min((int)floor(w*V1size_x), XSIZE(rQ) -1);`